submissions/OlexiyDobroskok/OOP_exercise/index.js

```diff
@@ -0,0 +1,138 @@
+class Inhabitant {
+  constructor({
+    species = "No species",
+    name = "No name",
+    gender = "No gender",
+    saying = "Hello",
+    friends = [],
+  }) {
+    this.species = species;
+    this.name = name;
+    this.gender = gender;
+    this.saying = saying;
+    this.friends = friends;
+  }
+
+  addFriend(person) {
+    this.friends.push(person);
+  }
+
+  introduceYourSelf() {
+    if (this.friends.length === 0) {
+      this.friends = "Looking for friends!";
+      return `<strong>${this.saying}! My name is ${this.name}. Species: ${this.species}. Gender: ${this.gender}. Friends: ${this.friends} </strong>`;
+    }
+    const friendsName = this.friends.map((friend) => friend.name).join(", ");
+    return `<strong>${this.saying}! My name is ${this.name}. Species: ${this.species}. Gender: ${this.gender}. Friends: ${friendsName}. </strong>`;
+  }
```

Comment:
This fragment can be made DRY.
Optimize this method.

Reply:
> This fragment can be made DRY. Optimize this method.

done